### DISCUSSION TITLE
How do I make an RSS feed for a changelog?

### DISCUSSION THREAD

**Discussion by RealLucioCarreras** [datenstrom/community#660] [Ask a question] [answered]
Hello,
first of all thank you very much for this project. I started two days ago with great success so far. But I have an understanding problem regarding feeds. Sorry for the long post ;)

## What I have:
* I have a directory `content/changelogs`
* Inside this directory there's a `page.md` and `changelog-v1.md`, `changelog-v2.md` and so on.
a changelog entry roughly looks like this:
```
---
Title: v1
Published: 2019-04-19
Author: Me
Layout: blog
Tag: Changelog
---
Some text
```

## What I try to achieve:
* Create a xml file which can be used in a rss feed reader

### try to set a custom feed directory
```
FeedLocation: /content/changelog/
FeedFileXml: feed.xml
FeedFilterLayout: blog
```
### copy the changelog directory to `<yellow-root>/feed`
```
FeedLocation: /feed/
FeedFileXml: feed.xml
FeedFilterLayout: blog
```
### copy the changelog directory to `<yellow-root>/content/feed`
```
FeedLocation: /feed/
FeedFileXml: feed.xml
FeedFilterLayout: blog
```
No matter what, but `http://localhost:8000/feed/page:feed.xml`, `http://website.org/feed/page:feed.xml` always result in `404`. `php yellow.php build` also never creates a `xml` file.

Is this even possible?

Thanks in advance.

**Comment by Acorn2Oak** [comment]
Have you checked out the [Feed](https://github.com/datenstrom/yellow-extensions/tree/master/source/feed) extension.
It creates xml in ../feed/page:feed.xml, which can be tweaked in system.ini as described.

**Reply by RealLucioCarreras** [reply]
Hello,
yes, I have. I played around with it for a few hours but I could not figure out how it works. So I decided to ask here. I cannot get any xml file. The `http://localhost:8000/feed` site is there, it's exactly the same html as in `http://localhost:8000/changelog`. Not surprising because I copied everything from `changelog` to `feed`. But `http://localhost:8000/feed/page:feed.xml` results in 
```
File not found
The requested file was not found. Oh no...
```
And no, there's no xml file generated in the entire source tree.

**Reply by markseuffert** [reply]
Hello Michael,

I might know what's going on. The problem you describe might come from combining blog and feed in one location. Datenstrom Yellow has a modular design, some extensions need their own location to work. The blog extension is invoked via `Layout: blog` setting and the feed extension via `Layout: feed` setting. When combining them in one location they would compete for attention. I recommend to keep the [original file](https://raw.githubusercontent.com/annaesvensson/yellow-feed/main/page.md) in `content/feed/page.md`. 

You can also develop your own changelog extension, there's an API for developers.  Hope this helps. 😁

**Reply by RealLucioCarreras** [reply]
Yes, this also solved my problem, thanks. Website is looking great now and works as intended. Thanks for your great project. You don't want me to program php, but do you have a donation button anywhere? You really saved me a lot of time.

**Reply by markseuffert** [reply]
Good luck with your website. No we don't take donations. You're welcome to contribute and give feedback: https://datenstrom.se/yellow/help/contributing-guidelines

**Comment by RealLucioCarreras** [comment, marked as answer]
Ok, I figured it out. I had to create a `page.md` within the `feed` directory 
```
---
Title: Feed
TitleSlug: Feed
Layout: feed
Status: unlisted
---
```

But thank you! :)
